### PR TITLE
[skiprutime-server] Respond with 404 error on UnknownCollectionError

### LINKED ISSUE #395: [skipruntime/server] We should respond with `404 NOT FOUND` for `UnknownCollectionError`.
We should respond with `404 NOT FOUND` for `UnknownCollectionError`.

_Originally posted by @beauby in https://github.com/SkipLabs/skdb/pull/389#discussion_r1795228445_

### PR DESCRIPTION
**Pull request by skiplabsdaniel**
Fix #395

### CODE REVIEW

File: skipruntime-ts/server/src/rest.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export function createRESTServer(runtime: SkipRuntime): express.Express {
       res.status(200).json({});
     } catch (e: unknown) {
       if (e instanceof UnknownCollectionError) {
-        res.status(400).json("Bad request");
+        res.status(404).json("Not Found");
```
Comment (beauby) [inline skipruntime-ts/server/src/rest.ts]: We should probably omit the body as a `404` response status implies the "Not found" information.

File: skipruntime-ts/core/src/skip-runtime.ts [hunk 2]
```diff
@@ -22,7 +22,10 @@ export type {
   SubscriptionID,
 } from "./skipruntime_api.js";
 
-export { UnknownCollectionError } from "./skipruntime_errors.js";
+export {
+  UnknownCollectionError,
+  AccessRefusedError,
```
Comment (beauby) [inline skipruntime-ts/core/src/skip-runtime.ts]: Is this needed?

File: skipruntime-ts/server/src/rest.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ export function createRESTServer(runtime: SkipRuntime): express.Express {
       );
       const [name, value] = reactiveResponseHeader(data);
       res.set(name, value);
-      res.status(200).json({});
+      res.status(200).json();
```
Comment (beauby) [inline skipruntime-ts/server/src/rest.ts:26]: I think you can simply remove the .json() call

Comment (skiplabsdaniel) [reply skipruntime-ts/server/src/rest.ts:26]: Yes but I need sendStatus instead of status.

**Pull request merged**